java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/utils/ClientAuthenticationUtils.java

```diff
@@ -101,6 +106,117 @@ private static CredentialCallOption getAuthenticate(final FlightClient client,
 
+  private static KeyStore getKeyStoreInstance(String instance)
+      throws KeyStoreException, CertificateException, IOException, NoSuchAlgorithmException {
+    KeyStore keyStore = KeyStore.getInstance(instance);
+    keyStore.load(null, null);
+
+    return keyStore;
+  }
+
+  private static String getOperatingSystem() {
+    return System.getProperty("os.name");
+  }
+
+  /**
+   * Check if the operating system running the software is Windows.
+   *
+   * @return whether is the windows system.
+   */
+  public static boolean isWindows() {
+    return getOperatingSystem().contains("Windows");
+  }
+
+  /**
+   * Check if the operating system running the software is Mac.
+   *
+   * @return whether is the mac system.
+   */
+  public static boolean isMac() {
+    return getOperatingSystem().contains("Mac");
+  }
+
+  /**
+   * Check if the operating system running the software is Linux.
+   *
+   * @return whether is the linux system.
+   */
+  public static boolean isLinux() {
+    return getOperatingSystem().contains("Linux");
+  }
+
+  /**
+   * Gets the trusted certificates input stream from the JVM.
+   *
+   * @return the trusted certificates as an {@link InputStream}.
+   * @throws IOException                if it fails reading the file.
+   * @throws KeyStoreException          if a key store could not be loaded.
+   * @throws CertificateException       if a certificate could not be found.
+   */
+  public static InputStream getCertificateInputStreamFromJVM()
+      throws IOException, KeyStoreException, CertificateException, NoSuchAlgorithmException {
+    Path path = Paths.get(System.getProperty("java.home"), "lib", "security", "cacerts");
+    InputStream fileInputStream = Files.newInputStream(path);
+    KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+    String password = "changeit";
+
+    keyStore.load(fileInputStream, password.toCharArray());
+
+    return getCertificatesInputStream(Collections.singletonList(keyStore));
+  }
+
+  /**
+   * It gets the trusted certificate based on the operating system and loads all the certificate into a
+   * {@link InputStream}.
+   *
+   * @return An input stream with all the certificates.
+   *
+   * @throws KeyStoreException        if a key store could not be loaded.
+   * @throws CertificateException     if a certificate could not be found.
+   * @throws IOException              if it fails reading the file.
+   */
+  public static InputStream getCertificateInputStreamFromSystem() throws KeyStoreException,
+      CertificateException, IOException, NoSuchAlgorithmException {
+
+    List<KeyStore> keyStoreList = new ArrayList<>();
+    if (isWindows()) {
+      keyStoreList.add(getKeyStoreInstance("Windows-ROOT"));
+      keyStoreList.add(getKeyStoreInstance("Windows-MY"));
+    } else if (isMac()) {
+      keyStoreList.add(getKeyStoreInstance("KeychainStore"));
+    } else {
+      throw new RuntimeException(getOperatingSystem() + "not a supported operating system");
```

Comment:
Just use the JDK default (and log as such) in this scenario.

Reply:
Fixed, I will create a ticket to add the log since we don't have any log in the JDBC